text/0510-dynamodb-global-table.md

```diff
@@ -0,0 +1,1742 @@
+  });
+```
+
+__NOTE__:
+
+* If the encryption mode is CUSTOMER_MANAGED, then you would need to specify a KMS key for each replica.
+
+#### Streams
+
+You can enable DynamoDB streams for your global tables. There are four possible values for it,
+
+1. `NEW_IMAGE`: The entire item, as it appears after it was modified, is written to the stream.
+2. `OLD_IMAGE`: The entire item, as it appeared before it was modified, is written to the stream.
+3. `NEW_AND_OLD_IMAGES`: Both the new and the old item images of the item are written to the stream.
+4. `KEYS_ONLY`: Only the key attributes of the modified item are written to the stream.
+
+This would be the same for each replica. By default, streams are not enabled for global table. But, if
+you have a replica present which is not of the same region as where the stack is deployed to, then the
+stream type would be set to `NEW_AND_OLD_IMAGES` automatically since it is required for table replication.
+
+```typescript
+  new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    stream: StreamViewType.NEW_IMAGE,
+  });
+```
+
+#### Time To Live
+
+You can enable time to live for an attribute of the global table. This would be the same for each replica.
+By default, it is not enabled for any attributes.
+
+```typescript
+  new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    timeToLiveAttribute: 'Foo',
+  });
+```
+
+#### Replicas
+
+You can define replicas for your global table. By default, if this configuration is undefined, then the
+table is deployed to the region where stack is being deployed.
+
+```typescript
+  const globalTable = new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    replicas: [
+      {
+        region: 'us-east-1',
+      },
+      {
+        region: 'us-west-2',
+      },
+    ],
+  });
+```
+
+There are per replica configurations that are available. If any of these are undefined in the replica and
+defined in the table level props, then those are copied over to the replica configuration.
+But, if the replica has any configuration that is defined and the tabel level properties are defined as well,
+then the replica property value takes precedence and would be assigned.
+
+Options:
+
+* `region` --> Needs to be specified
+* `contributorInsightsEnabled` --> Gets copied over from table level props if defined.
+* `deletionProtection` --> Gets copied over from table level props if defined.
+* `pointInTimeRecovery` --> Gets copied over from table level props if defined.
+* `tableClass` --> Gets copied over from table level props if defined.
+* `tags` --> Gets copied over from table level props if defined.
+* `readWithAutoScaling` --> Gets copied over from table level props if defined.
+* `readCapactity` --> Gets copied over from table level props if defined.
+* `encryptionKey` --> This needs to be defined per replica
```

Comment:
This might be tricky to get. We need to get the ARN of a key in a different region.